toolbox/metrics/main.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"golang.org/x/net/context"
+)
+
+const (
+	fetchInterval = 5
+)
+
+var (
+	webAddress = flag.String("listen_port", ":9103", "port on which to expose metrics and web interface.")
+	jenkinsURL = flag.String("jenkins_url", "https://testing.istio.io", "URL to Jenkins API")
+	gcsBucket  = flag.String("bucket", "istio-code-coverage", "gcs bucket")
+
+	succeededBuilds = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: "succeeded_build_durations_seconds",
+			Help: "Succeeded build durations seconds.",
+		},
+		[]string{"build_job", "repo"},
+	)
+
+	failedBuilds = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: "failed_build_durations_seconds",
+			Help: "Failed build durations seconds.",
+		},
+		[]string{"build_job", "repo"},
+	)
+
+	succeededBuildMax = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "succeeded_build_durations_max_seconds",
+			Help: "Succeeded max build durations seconds.",
+		},
+		[]string{"build_job", "repo"},
+	)
+
+	succeededBuildMin = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "succeeded_build_durations_min_seconds",
+			Help: "Succeeded min build durations seconds.",
+		},
+		[]string{"build_job", "repo"},
+	)
+
+	codeCoverage = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "package_code_coverage",
+			Help: "Package code coverage.",
+		},
+		[]string{"package", "repo"},
+	)
+
+	jobRegister = map[string][]string{
+		"pilot":       {"presubmit", "postsubmit", "e2e-smoketest", "e2e-suite"},
+		"mixer":       {"presubmit", "postsubmit", "e2e-smoketest", "e2e-suite"},
+		"proxy":       {"presubmit", "postsubmit"},
+		"auth":        {"presubmit", "postsubmit"},
+		"mixerclient": {"presubmit"},
+		"istio":       {"presubmit"},
+		"test-infra":  {"presubmit"},
+	}
+
+	codeCovRegister = map[string]bool{
```

Comment:
Same comment here, Can we attempt to get code coverage on all the repos and just print 0 on others ?